specification/network/resource-manager/Microsoft.Network/2017-06-01/networkInterface.json

```diff
@@ -579,11 +607,11 @@
         },
         "protocol": {
           "type": "string",
-          "description": "The network protocol this rule applies to. Possible values are: 'Tcp', 'Udp', and '*'.",
+          "description": "The network protocol this rule applies to. Possible values are: 'Tcp', 'Udp', and 'All'.",
           "enum": [
             "Tcp",
             "Udp",
-            "*"
+            "All"
```

Comment:
@jianghaolu, I think these should be independent types. I've checked both and [this](https://github.com/EvgenyAgafonchikov/azure-rest-api-specs/blob/43acc9dd8542870186bfbd91e8980aaf7f227bf4/specification/network/resource-manager/Microsoft.Network/2017-06-01/networkSecurityGroup.json#L477) returns "*" as protocol of `SecurityRule` in server response while `EffectiveNetworkSecurityRule` has "All" in response